File: Python3/Hard/RuntimeError/stamping-the-sequence_7.py
class Solution:
    def movesToStamp(self, stamp: str, target: str) -> List[int]:
        
        # replace target with ?
        target = list(target)
        n = len(target)
        # replace target with ?
        for i in range(n):
            if target[i] == stamp[0]:
                target[i] = '?'
        # print(target)
        # make a dict of stamp
        stamp_dict = {}
        for i in range(len(stamp)):
            if stamp[i] not in stamp_dict:
                stamp_dict[stamp[i]] = [i]
            else:
                stamp_dict[stamp[i]].append(i)
        # print(stamp_dict)
        # start from the first index of stamp
        # if the char is ?, then find the index of the char in stamp
        # replace the char at the index with ?
        # if the char is not ?, then find the index of the char in stamp
        # replace the char at the index with ?
        # if the char is not ?, then find the index of the char in stamp
        # replace the char at the index with ?
        # if the char is not ?, then find the index of the char in stamp
        # replace the char at the index with ?
        # if the char is not ?, then find the index of the char in stamp
        # replace the char at the index with ?
        # if the char is not ?, then find the index of the char in stamp
        # replace the char at the index with ?
        # print(target)
        # print(stamp)
        # print(stamp_dict)
        result = []
        def dfs(target, stamp, stamp_dict, index, result):
            # print(target)
            # print(stamp)
            # print(stamp_dict)
            # print(index)
            # print(result)
            # print()
            if index == len(stamp):
                return True
            if target[index] == '?':
                # print('here')
                # print(target)
                # print(stamp)
                # print(stamp_dict)
                # print(index)
                # print(result)
                # print()
                if stamp[index] in stamp_dict:
                    # print('here')
                    # print(target)
                    # print(stamp)
                    # print(stamp_dict)
                    # print(index)
                    # print(result)
                    # print()
                    for i in stamp_dict[stamp[index]]:
                        # print('here')
                        # print(target)
                        # print(stamp)e last movis

File: Java/Hard/RuntimeError/stamping-the-sequence_7.java
class Solution {
    public int[] movesToStamp(String stamp, String target) {
        
        int[] ans = new int[1000];
        int t = 0;
        char[] cs = stamp.toCharArray();
        char[] ct = target.toCharArray();
        boolean[] done = new boolean[ct.length];
        int count = 0;
        while (count < ct.length) {
            boolean found = false;
            for (int i = 0; i <= ct.length - cs.length; i++) {
                if (done[i]) {
                    continue;
                }
                int stars = 0;
                int changed = 0;
                for (int j = 0; j < cs.length; j++) {
                    if (ct[i + j] == '*') {
                        stars++;
                    } else if (ct[i + j] == cs[j]) {
                        changed++;
                    }
                }
                if (stars + changed == cs.length) {
                    for (int j = 0; j < cs.length; j++) {
                        if (ct[i + j] != '*') {
                            ct[i + j] = '*';
                            count++;
                        }
                    }
                    ans[t++] = i;
                    found = true;
                }
            }
            if (!found) {
                return new int[0];
            }
        }
        int[] ret = new int[t];
        for (int i = 0; i < t; i++) {
            ret[i] = ans[t - i - 1];
        }
        return ret;
    }
}